### DISCUSSION TITLE
Crash SectionIsLeakingFromOutside

### DISCUSSION THREAD

**Discussion by benjamincrane** [FakeFishGames/Barotrauma#13945] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

During a multiplayer game that I was hosting, while at least one player was outside the submarine, the application crashed. I have attached the crash reports and server logs. It happened twice at different times and places in the same level.

The error message that might be the problem says SectionIsLeakingFromOutside.

Mods being used:
Selected content packages: Vanilla (329B29B), Harmonica Free Bird solo (96C6D78), Detectable Alien Minerals (41559E3), Plant Scanner Vanilla (ABED508), Fortunate Worm (E6C5059), Attack Submarine Mineral Scanner Upgrade (6F23B3C)

[crashreport.log](https://github.com/FakeFishGames/Barotrauma/files/15257303/crashreport.log)
[crashreport (2).log](https://github.com/FakeFishGames/Barotrauma/files/15257304/crashreport.2.log)
[clontarf_2024-05-08_2201 (2).txt](https://github.com/FakeFishGames/Barotrauma/files/15257311/clontarf_2024-05-08_2201.2.txt)
[clontarf_2024-05-08_2201.txt](https://github.com/FakeFishGames/Barotrauma/files/15257312/clontarf_2024-05-08_2201.txt)
[clontarf_2024-05-08_2336.txt](https://github.com/FakeFishGames/Barotrauma/files/15257313/clontarf_2024-05-08_2336.txt)

### Reproduction steps

It seems to happen when players are outside the submarine and submarine turrets are firing?

### Bug prevalence

Twice in the same night

### Single player or multiplayer?

Multiplayer hosted from the in-game menu (= using a listen server)

### Version

v1.4.6.0 (Blood in the Water Update, hotfix 2)

### Which operating system did you encounter this bug on?

Windows

### Relevant error messages and crash reports
Barotrauma seems to have crashed. Sorry for the inconvenience! 

Exception: Object reference not set to an instance of an object. (System.NullReferenceException)
Target site: Boolean SectionIsLeakingFromOutside(Int32)
Stack trace: 
   at Barotrauma.Structure.SectionIsLeakingFromOutside(Int32 sectionIndex) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Map\Structure.cs:line 930
   at Barotrauma.Items.Components.RepairTool.FixBody(Character user, Vector2 hitPosition, Single deltaTime, Single degreeOfSuccess, Body targetBody) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Items\Components\Holdable\RepairTool.cs:line 571

**Comment by benjamincrane** [comment]
Log from computer of one of the other players.

[crashreport_2.log](https://github.com/FakeFishGames/Barotrauma/files/15283914/crashreport_2.log)

**Comment by Regalis11** [comment]
Duplicate of https://github.com/FakeFishGames/Barotrauma/discussions/13869
